### ISSUE TITLE
Login dialog doesn't close after login

### ISSUE THREAD

**Issue by edparsons** [jeduan/cordova-plugin-facebook4#190]
After the app is successfully logged in, I expect the SafariView to close. Instead the view remains on the screen until the user clicks "Done".

Below is the Xcode output:

2016-04-04 13:14:43.009 tapcast_mobile[72410:2870196] -canOpenURL: failed for URL: "fbauth2://" - error: "This app is not allowed to query for scheme fbauth2"
2016-04-04 13:14:43.024 tapcast_mobile[72410:2870196] THREAD WARNING: ['FacebookConnectPlugin'] took '21.370117' ms. Plugin should use a background thread.
2016-04-04 13:16:07.950 tapcast_mobile[72410:2870196] -canOpenURL: failed for URL: "fbauth2://" - error: "This app is not allowed to query for scheme fbauth2"
2016-04-04 13:16:10.866 tapcast_mobile[72410:2870196] {"status":"connected","authResponse":{"secret":"...","session_key":true,"sig":"...","accessToken":"EAALTk0RbSYwBABTqCeHL6LIsDhdSuxaiHZAAbZARyP7kB9srZAHlriInVAnlYwxFVaEAspirRXh1ZChGbOjLVa9IvnndNtCgFikWktDnKdBo8xRUWzvupjN0cNGJOiPeaQBsCO7q6DSSNKFAz9nljRyQ7FINrtvpCQKtKwNTvqZADOBERFvYtAdWTWqENIhEdgjVtW5EKEEVjeZAcvid8ifgjZCJdz7a6QZD","userID":"10153333146415974","expiresIn":"5182686"}}

**Comment by vivekpd15** [comment]
I am experiencing the same issue on iOS9. Works perfectly on Android though. Any solution in sight yet?

**Comment by Hirbod** [comment]
Can't confirm this. I've tested multiple times. Looks like there is something wrong setup. canOpenUrl errors should be a hint

**Comment by mattpowell** [comment]
I'm currently running in to this issue as well using the latest version of cordova-plugin-facebook4 on iOS 9.2. The issue doesn't seem to be present on iOS 9.1.

There's an additional log after the canOpenUrl messages that may be helpful:
```
2016-04-29 15:37:27.982 AppNameHere[42432:19656076] Warning: Attempt to present <FBSDKContainerViewController: 0x7fe2b24c7de0> on <MainViewController: 0x7fe2b26438c0> whose view is not in the window hierarchy!
```

**Comment by conor-mac-aoidh** [comment]
I'm seeing this issue too on iOS 9.2. +1

**Comment by robertarnesson** [comment]
what version of the plugin are you using?

**Comment by mattpowell** [comment]
@robertarnesson I was using v1.7.1. I also spent a little bit of time trying to debug the issue locally, but, w/o much avail. Tried upgrading to the latest version of the Facebook SDK, reverting commit 4c6d3ee4b7dce81d26e77d1d162231e70351a6e0 (since my error message seemed sort of related), and upgrading/removing various other plugins. Nothing worked, but, admittedly my Objective-C/etc is a bit lacking :/
